docs/source/daml/intro/6_Parties.rst

```diff
@@ -0,0 +1,197 @@
+
+.. literalinclude:: daml/daml-intro-6/Parties.daml
+  :language: daml
+  :start-after: -- SENDER_ROLE_BEGIN
+  :end-before: -- SENDER_ROLE_END
+
+The above ``IouSender`` contract now gives a one party, the ``sender`` the right to send ``Iou`` contracts with positive amounts to a ``receiver``. The ``nonconsuming`` keyword on the choice ``Send_Iou`` changes the behaviour of the choice so that the contract it's exercised on does not get archived when the choice is exercised. That way the ``sender`` can use the contract to send multiple Ious. Here it is in action:
+
+.. literalinclude:: daml/daml-intro-6/Parties.daml
+  :language: daml
+  :start-after: -- SENDER_SCENARIO_BEGIN
+  :end-before: -- SENDER_SCENARIO_END
+
+DAML's authorization model
+--------------------------
+
+Hopefully, the above will have given you a good intuition for how authority is passed around in DAML. In this section you'll learn about  the formal authorization model to allow you to reason through your contract models. This will allow you to construct them in such a way that you don't run into authorization errors at runtime, or, worse still, allow malicious transactions.
+
+In :ref:`choices` you learned that a transaction is, equivalently, a tree of transactions, or a forest of actions, where each transaction is a list of actions, and each action has a child-transaction called its consequences.
+
+Each action has a set of *required authorizers*:
+
+- The required authorizers of an exercise action are specified as the controllers on the corresponding choice. Remember that ``Archive`` and ``archive`` are just an implicit choice with the signatories as controllers.
+- The required authorizers of a create action are the signatories of the contract.
+- The required authorizers of a fetch (which also includes ``fetchByKey``) are somewhat dynamic and covered later.
+
+Each transaction has a set of *authorizers*:
+
+- The root transaction of a commit is authorized by the submitting party.
+- The consequences of an exercise action are authorized by the actors of that action plus the signatories of the contract on which the action was taken.
+
+The authorization rule is that the required authorizers of every action are a subset of the authorizers of the parent transaction.
+
+Take, for example, the final transaction in the scenario of the the source file for this section. Ignoring fetches:
+
+- Bob submits the transaction so he's the authorizer on the root transaction.
+- The root transaction has a single action, which is to exercise ``Send_Iou`` on a ``IouSender`` contract with Bob and ``sender`` and Charlie as ``receiver``. Since the controller of that choice is the ``sender``, Bob is the required authorizer.
```

Comment:
```suggestion
- The root transaction has a single action, which is to exercise ``Send_Iou`` on a ``IouSender`` contract with Bob as ``sender`` and Charlie as ``receiver``. Since the controller of that choice is the ``sender``, Bob is the required authorizer.
```